src/rebar3_hex_docs.erl

```diff
@@ -118,23 +142,32 @@ do_revert(App, State, Repo, Vsn) ->
             ?PRV_ERROR({revert, Reason})
     end.
 
-file_list(Files) ->
-    [{drop_path(ShortName, ["doc"]), FullName} || {ShortName, FullName} <- Files].
+%% @doc Returns the directory were docs are to be found
+%%
+%% The priority for resolution is the following:
+%%   1. `doc' entry in the application's `*.app.src'.
+%%   2. `dir' entry specified in `edoc_opts'.
+%%   3. `"doc"' fallback default value.
+-spec resolve_doc_dir(rebar_app_info:t()) -> string().
+resolve_doc_dir(AppInfo) ->
+    AppOpts = rebar_app_info:opts(AppInfo),
+    EdocOpts = rebar_opts:get(AppOpts, edoc_opts, []),
+    AppDetails = rebar_app_info:app_details(AppInfo),
+    Dir = proplists:get_value(dir, EdocOpts, ?DEFAULT_DOC_DIR),
+    proplists:get_value(doc, AppDetails, Dir).
+
+-spec assert_doc_dir(string()) -> true.
+assert_doc_dir(DocDir) ->
+    filelib:is_dir(DocDir) orelse
+        rebar_api:abort( "Docs were not published since they "
```

Comment:
Since the `publish` command call the `rebar3_hex_docs:publish/3` function once it has already succeeded publishing the package itself, I though it would be better to use `rebar_api:abort/2` for showing the user the error and aborting the execution.

Reply:
Yeah I wonder if we actually want to bail on the publish 🤔 Like, if you're expecting it all in one go, then I could see where we might want to bail on the package publishing too, but I don't have strong feelings about it either.

Reply:
Yeah, I thought about this more and I really think we should strive for parity with hex (mix hex) here. Specifically, calling `rebar3 hex publish` assumes you want to publish both, we should check that docs exist before we attempt to publish anything, if they do not, bail and give the user a message.

We should also provide switches so inform publish to only publish the package or only publish the docs. `--package` and `--docs`. 

Thoughts? 

Likewise, there should IMO be a switches to tell publish to do one or the other `rebar3 hex publish --only-package` or `--package` same for docs. Users can optionally use

Reply:
I think that is a good idea, but I think the discussion around the details of implement this new approach deserves its own issue and separate PR. There might be many details that need to be discusses, for example, it would be nice to show the docs directory that will be used when listing the details on what is going to be published. 

Since this PR address the bug in the current behaviour discussed in #180 I would like to get it merged to avoid the unexpected results we get today.

Reply:
To be clear, since I'm missing some context, we are running the `edoc` provider before publishing, right? If not we should have publish depending on or calling the edoc provider from rebar3 and then doing the publish.

If there aren't any docs even though it ran edoc then it should just continue on I would think, without erroring. Obviously will be an issue of someone overriding where edoc outputs shit, but whatever, can deal with that later.

Reply:
The `publish` command currently also publishes the docs by calling the internal function in `rebar3_hex_docs:publish3`. It is in this situation that the published docs are empty.

Reply:
Ah ok. Yea, I guess that function should call the edoc provider instead of having the provider use a dependency on it.

Reply:
Yeah, I'm cool with merging this now per what @jfacorro points out, i.e., it fixes a very confusing issue.